pex/interpreter.py

```diff
@@ -320,67 +329,73 @@ def expand_path(cls, path):
   def from_env(cls, hashbang):
     """Resolve a PythonInterpreter as /usr/bin/env would.
 
-       :param hashbang: A string, e.g. "python3.3" representing some binary on the $PATH.
+    :param hashbang: A string, e.g. "python3.3" representing some binary on the $PATH.
+    :return: the first matching interpreter found or `None`.
```

Comment:
> the first matching interpreter

I want to ensure that interpreter discovery is deterministic, but I'm having a hard time reasoning about how we'd verify that. Could you please speak to your expectations around determinism of interpreter resolution when there are multiple matching candidates?

Reply:
Everything funnels to `_identify_interpreters` which deterministically iterates interpreters based on the order of an explicit `paths` sequence or else the sequence defined by the `PATH` env var. If that paths input is stable, the resulting iteration order of interpreters is too.